### ISSUE TITLE
Add Adaptive Cruise Controller (ACC) model

### ISSUE THREAD

**Issue by mforets** [JuliaReach/NeuralNetworkAnalysis.jl#2]

**Comment by SebastianGuadalupe** [comment]
WIP model:
```julia
@taylorize function ACC!(du, u, p, t)
    local f, a_lead, a_ego = 0.0001, -2., -1.
    # u[1] = x_lead
    # u[2] = v_lead
    # u[3] = γ_lead
    # u[4] = x_ego
    # u[5] = v_ego
    # u[6] = γ_ego
    
    du[1] = u[2]
    du[2] = u[3]
    du[3] = -2 * u[3] + 2 * a_lead - 2 * f * u[2] * u[2]
    du[4] = u[5]
    du[5] = u[6]
    du[6] = -2 * u[6] + 2 * a_ego - 2 * f * u[5] * u[5]
    return du
end

# define the initial-value problem
X₀ = Hyperrectangle(low=[90., 32, 0, 10, 30, 0], high=[110, 32.2, 0, 11, 30.2, 0])

prob = @ivp(x' = ACC!(x), dim: 6, x(0) ∈ X₀)

# solve it
@time sol = solve(prob, T=0.1);
```

**Comment by mforets** [comment]
good! the rhs of `u[3]` does not satisfy the rule of only using binary operations (see [this paragraph](https://juliareach.github.io/ReachabilityAnalysis.jl/dev/man/nonlinear/#How-can-I-get-the-most-out-of-@taylorize?-1)), s i would suggest to re-write this as

```julia
    two_f = 2 * f
    u2sq = u[2] * u[2]
    u5sq = u[5] * u[5]

    du[1] = u[2]
    du[2] = u[3]
    du[3] = -2 * u[3] + (2 * a_lead - two_f * u2sq)
    du[4] = u[5]
    du[5] = u[6]
    du[6] = -2 * u[6] + (2 * a_ego - two_f * u5sq)
```

**Comment by SebastianGuadalupe** [comment]
Fixed by #6.
